core/pta/attestation.c

```diff
@@ -53,7 +55,28 @@ static int cmp_regions(const void *a, const void *b)
 	return memcmp((void *)r1->va, (void *)r2->va, r1->size);
 }
 
-static TEE_Result hash_regions(struct vm_info *vm_info, uint8_t *hash)
+static TEE_Result hash_update(void *ctx, struct vm_region *r, uint32_t mode)
+{
+	struct file *f = NULL;
+	uint8_t *tag = NULL;
+
+	switch (mode) {
+	case PTA_ATTESTATION_HASH_MODE_FULL:
+		return crypto_hash_update(ctx, (uint8_t *)r->va, r->size);
+	case PTA_ATTESTATION_HASH_MODE_TAGS:
+		f = to_file_may_fail(r->mobj);
+		/* Regions we're interested in should all have a struct file */
+		assert(f);
+		tag = file_get_tag(f);
+		assert(tag);
+		return crypto_hash_update(ctx, tag, FILE_TAG_SIZE);
```

Comment:
This region probably only maps a part of the file, so we should try to dig out the offset into the file used and the size of this region and hash those values too.

Reply:
Updated, not quite sure of my code though (locking?)